### ISSUE TITLE
In the local distributed environment, can not create table

### ISSUE THREAD

**Issue by Fengys123** [GreptimeTeam/greptimedb#1205]
### What type of bug is this?

Unexpected error

### What subsystems are affected?

Datanode

### What happened?

In the local distributed environment, the table cannot be created. The error is as follows:
```
mysql> CREATE TABLE dist_table(
    ->     ts TIMESTAMP DEFAULT current_timestamp(),
    ->     n INT,
    ->     row_id INT,
    ->     TIME INDEX (ts)
    -> )
    -> PARTITION BY RANGE COLUMNS (n) (
    ->     PARTITION r0 VALUES LESS THAN (5),
    ->     PARTITION r1 VALUES LESS THAN (9),
    ->     PARTITION r2 VALUES LESS THAN (MAXVALUE),
    -> )
    -> engine=mito;
ERROR 1815 (HY000): Failed to execute query: CREATE TABLE dist_table(
    ts TIMESTAMP DEFAULT current_timestamp(),
    n INT,
    row_id INT,
    TIME INDEX (ts)
)
PARTITION BY RANGE COLUMNS (n) (
    PARTITION r0 VALUES LESS THAN (5),
    PARTITION r1 VALUES LESS THAN (9),
    PARTITION r2 VALUES LESS THAN (MAXVALUE),
)
engine=mito, source: Failed to execute statement, source: Failure during query execution, source: Failed to request Meta, source: Illegal state from server, code: 1, error: No active datanodes
```

### What operating system did you use?

MacBook Pro M1

### How can we reproduce the bug?

```bash
I followed the steps of the documentation. 
https://docs.greptime.com/getting-started/distributed

But the command to start meta is different. 
cargo run -- metasrv start --selector LoadBased
```

**Comment by evenyag** [comment]
@Fengys123 What's the root cause of this bug?

**Comment by Fengys123** [comment]
In the logic of heartbeat reporting, due to some problems in error handling, datanode reports the region-number of heartbeat is -1.

If the [region number](https://github.com/GreptimeTeam/greptime-proto/blob/main/proto/greptime/v1/meta/heartbeat.proto#L45) in the heartbeat sent by datanode to meta is -1, it means that something went wrong. In the current implementation of [LoadbasedSelector](https://github.com/GreptimeTeam/greptimedb/blob/develop/src/meta-srv/src/selector/load_based.rs), it relies on the region number in the heartbeat. If the region number is equal to -1, LoadbasedSelector will exclude the datanode.

This issue has been resolved in #1207. But we may need to add some integration tests later.

**Comment by shuiyisong** [comment]
In #1167 `region_stats` was introduced to `table` trait with the default implementation of returning `UnsupportedSnafu`. During datanode starting up, `[numbers](https://github.com/GreptimeTeam/greptimedb/blob/b2a09c888a9e93a0b7ea6a78e87bb8ca8edaafb3/src/catalog/src/remote/manager.rs#L410)` table was registered as `NumebrTable` which doesn't explicitly implement `region_stats`. 

When heartbeat task tries to iterate over tables to collect region stats, `numbers` table would fail, stopping the whole collecting process and return `-1` as the final result. Then the datanode is excluded by the meta-srv as #Fengys123 explained above.
